### PR TITLE
Add reduction_unroll_factor to autotuning script

### PR DESCRIPTION
**Pull request by rdspring1**
This PR renames `unroll_factor` to `iteration_unroll_factor` and adds `reduction_unroll_factor`. `reduction_unroll_factor` adds unroll factor on top of vectorization factor for the inner reduction domain.

### CODE REVIEW

**Comment by rdspring1** [comment]
!test

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: doc/dev/python_scheduling/autotune_inner_reduction.py [hunk 1]
```diff
@@ -184,16 +202,16 @@ def generate_scheduler_configurations(self, input_shape):
...
 
             # number of reduction elements not handled by a CTA
             remaining_reduction = ceil_div(
                 num_reductions,
-                (scheduler_config.bdimx * scheduler_config.vectorize_factor),
+                (scheduler_config.bdimx * vectorize_factor * reduction_unroll_factor),
             )
 
-            if unroll_factor == 1 and remaining_reduction > 1:
+            if iteration_unroll_factor == 1 and remaining_reduction > 1:
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply doc/dev/python_scheduling/autotune_inner_reduction.py:214]: From `inner2dReductionHeuristic`, I see this:

```cpp
  // Cross grid reduction if we haven't hit our target blocks, and we have manyr
  // reduction elements.
  if ((godim < target_blocks && remainder_in_reduction >= 0) ||
      (remainder_in_reduction >= kEight)) {
    grdim = remainder_in_reduction;
  }

    // Try to do some cleanup of ragged waves on device
    { do_something }

    // Grid reductions do not support unrolling iteration dimension, revert if
    // set. Recalculate godim.
    { do_something }
```

[the remark this answers is not in the document]
Comment (liqiangxl) [reply doc/dev/python_scheduling/autotune_inner_reduction.py:216]: another approch is we can add another search para `is_block_reduction`, if it is true, we only use block reduction, if it is false, we do grid reduction.

**Review by liqiangxl — APPROVED** [review]
LGTM.

**Comment by rdspring1** [comment]
!build

**Pull request merged**